core/trino-main/src/main/java/io/trino/execution/QueryStats.java

```diff
@@ -264,6 +266,7 @@ public QueryStats(
         this.peakTaskRevocableMemory = requireNonNull(peakTaskRevocableMemory, "peakTaskRevocableMemory is null");
         this.peakTaskTotalMemory = requireNonNull(peakTaskTotalMemory, "peakTaskTotalMemory is null");
         this.scheduled = scheduled;
+        this.faultTolerantStageScheduled = faultTolerantStageScheduled;
```

Comment:
This feels like a bit to much for this single usecase. Maybe instead let's have a flag if query is using FTE and a counter how many stages are already scheduled?

Reply:
I have thought about that. However, that may cause confusion to the end users as we won't be able to know whether we are using FTE here https://github.com/trinodb/trino/blob/49e4c06aff0889f84f69158aa5cf4b185da840e8/core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java#L282-L287

Reply:
I do not think it is a big deal that we always return `false` here. 
We can also make value OptionalBoolean. 
Or we can have a `String retryMode` field where we can put
`NONE`, `QUERY`, `TASK` and `UNKNOWN`. We can then use `UNKNOWN` in this context